### PR TITLE
TRAN-77: added Add and Edit modals to manage customer type

### PR DESCRIPTION
**Pull request by sniedzielski**
TICKET: https://sd-transit.atlassian.net/browse/TRAN-77

Peek 2022-09-02 15-24

### CODE REVIEW

File: src/components/forms/customerType/CustomerTypeForm.tsx [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+                value={{ className: 'float-right h-8 w-12 justify-end' }}
+              >
+                <RiCloseFill onClick={onCancel} />
+              </IconContext.Provider>
+            </div>
+            <div className="h-12">
+              <Input
+              // eslint-disable-next-line react/jsx-props-no-spreading
+                {...register('customerTypeName', { required: true })}
+                className=""
```
Comment (KarolinaKopacz) [inline src/components/forms/customerType/CustomerTypeForm.tsx]: Empty className is unnecessary.

File: src/pages/menuUpload/CustomerTypePage.tsx [hunk 2]
```diff
@@ -1,17 +1,39 @@
...
+import Dialog from '../../shared/dialog/Dialog';
 import PageBody from '../../components/shared/PageBody';
 import PageHeader from '../types';
 
 function CustomerTypePage() {
+  const [displayModal, setDisplayModal] = useState(false);
+  const [displayEditModal, setDisplayEditModal] = useState(false);
+  const [objectToEdit, setObjectToEdit] = useState({ id: null, customerTypeName: '' });
+
+  const toggleModal = () => {
```
Comment (KarolinaKopacz) [inline src/pages/menuUpload/CustomerTypePage.tsx]: Which modal? If it's modal for adding better change name to `toggleAddModal` :)

File: src/components/forms/customerType/CustomerTypeForm.tsx [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+import Input from '../../../shared/inputs/input';
+import ValidationError from '../../shared/ValidationError';
+import SubmitButton from '../../../shared/buttons/SubmitButton';
+import CancelButton from '../../../shared/buttons/CancelButton';
+
+interface CustomerTypeFormProps {
+  onSubmit: (formValues: FieldValues) => void;
+  onCancel: () => void;
+  title: string;
+  initialFormValue: any;
```
Comment (KarolinaKopacz) [inline src/components/forms/customerType/CustomerTypeForm.tsx]: These `initialFormValue` has not just object type?

File: src/pages/menuUpload/CustomerTypePage.tsx [hunk 4]
```diff
@@ -1,17 +1,39 @@
...
+    setDisplayModal(!displayModal);
+  };
+
+  const toggleEditModal = (object?:any) => {
```
Comment (KarolinaKopacz) [inline src/pages/menuUpload/CustomerTypePage.tsx]: These `object` has not just object type?

File: tailwind.config.js [hunk 5]
```diff
@@ -9,6 +9,7 @@ module.exports = {
       "transit-green": {
         DEFAULT: "#E0EEEF",
         dark: "#017E84",
+        light: "#E0EEEF",
```
Comment (KarolinaKopacz) [inline tailwind.config.js]: #E0EEEF this color is already defined in  `DEFAULT: "#E0EEEF"`. If you want to use it just write in className="..-transit-green"

**Review by KarolinaKopacz — CHANGES_REQUESTED** [review]

**Comment by sniedzielski** [comment]
@KarolinaKopacz fixed.

**Review by KarolinaKopacz — APPROVED** [review]

**Pull request merged**